src/windows/map.c

```diff
@@ -1349,6 +1349,12 @@ static void map_window_increase_map_size()
 	map_extend_boundary_surface();
 	window_map_init_map();
 	window_map_center_on_view_point();
+    // Recalculate fences
+	for (int y = 0; y < RCT2_GLOBAL(RCT2_ADDRESS_MAP_MAX_XY, uint16); y += 32) {
```

Comment:
I don't think this is really the right place, as it is game logic. I'd say it should be in `src/world/map.c`.
What do you think @IntelOrca ?